request.go

```diff
@@ -12,6 +13,9 @@ type Future struct {
 	requestId   uint32
 	requestCode int32
 	timeout     time.Duration
+	mutex       sync.RWMutex
+	cond        *sync.Cond
```

Comment:
Instead of (ab)using sync.Cond, you'd better
```
    // this doesn't consume more memory, than make(chan struct{})
    fut.ready := make(chan struct{}, 1000000000) 
```
And then notify about PUSH using `fut.ready <- struct{}{}`
This way you will need no separate goroutine to merry push notifications with timeouts and final response.
Whole iterator handling will be much simpler.

Reply:
Yes, it will be easy. But the code is similar to the Lua implementation will be incorrect:
https://www.tarantool.io/en/doc/latest/reference/reference_lua/box_session/push/
```Lua
for i, message in future:pairs() do
    table.insert(messages, message) table.insert(keys, i) end
messages
future:wait_result(1000)
for i, message in future:pairs() do
    table.insert(messages, message) table.insert(keys, i) end
messages
```

Reply:
Thank you, it's much easier.